great_expectations/data_context/types/base.py

```diff
@@ -148,14 +180,108 @@ class Meta:
 
     assets = fields.Dict(
         keys=fields.Str(),
-        values=fields.Nested(AssetConfigSchema),
+        values=fields.Nested(AssetConfigSchema, required=False, allow_none=True),
         required=False,
         allow_none=True,
     )
 
+    base_directory = fields.String(required=False, allow_none=True)
+    glob_directive = fields.String(required=False, allow_none=True)
+    default_regex = fields.Dict(required=False, allow_none=True)
+    runtime_keys = fields.List(
+        cls_or_instance=fields.Str(), required=False, allow_none=True
+    )
+    bucket = fields.String(required=False, allow_none=True)
+    prefix = fields.String(required=False, allow_none=True)
+    delimiter = fields.String(required=False, allow_none=True)
+    max_keys = fields.Integer(required=False, allow_none=True)
+    boto3_options = fields.Dict(
+        keys=fields.Str(), values=fields.Str(), required=False, allow_none=True
+    )
+    data_asset_name_prefix = fields.String(required=False, allow_none=True)
+    data_asset_name_suffix = fields.String(required=False, allow_none=True)
+    include_schema_name = fields.Boolean(required=False, allow_none=True)
+    splitter_method = fields.String(required=False, allow_none=True)
+    splitter_kwargs = fields.Dict(required=False, allow_none=True)
+    sampling_method = fields.String(required=False, allow_none=True)
+    sampling_kwargs = fields.Dict(required=False, allow_none=True)
+    excluded_tables = fields.List(
+        cls_or_instance=fields.Str(), required=False, allow_none=True
+    )
+    included_tables = fields.List(
+        cls_or_instance=fields.Str(), required=False, allow_none=True
+    )
+    skip_inapplicable_tables = fields.Boolean(required=False, allow_none=True)
+
     @validates_schema
     def validate_schema(self, data, **kwargs):
-        pass
+        # If a class_name begins with the dollar sign ("$"), then it is assumed to be a variable name to be substituted.
+        if data["class_name"][0] == "$":
+            return
+        if ("default_regex" in data) and not (
```

Comment:
@jcampbell @alexsherstinsky 
Because of this change, new custom DataConnector's cannot be constructed with `default_regex` and `glob_directive`. Any new custom custom data connectors that are constructed will raise errors here. Can I construct custom classes with default_regex and glob_directive somehow ?